### PR TITLE
RWIB Mission 2, Community Page

### PR DESCRIPTION
**Pull request by efindlay**
r? @brson
1. removed links on front page, and replaced with 'news', 'community link' and 'legal'
2. Major changes to Community page
3. Added How_to_Contribute_page, mainly based on http://www.ncameron.org/rust.html. It includes mostly links to issues.
4. Added User_Group page

### CODE REVIEW

**Comment by brson** [comment]
First impression looks good.

I like that you modified the navbar to include the community page until it gets its own redesign.

The 'legendary dedication' link: I was thinking something like http://techwhirl.com/conference-report-friendly-diverse-rust-camp/, though I really wanted like one glowing HN comment that could be directly linked. @steveklabnik do you have maybe a single tweet somewhere with glowing praise for the rust community?

The #rust-webdev and #rust-offtopic links are broken.

The description of #rust-internals includes the phrase 'for questions about how rust works'. I think this is too open to interpretation - we want general questions to go elsewhere, leaving internals for work. Maybe rephrase it to be more like 'To get involved of the development of Rust itself...'.

I'd also list #rust-internals second in the list, after #rust, since it's still sort of the primary entry point for internals discussion.

In addition to #rustc and #rustc-libs, I'd also suggest listing #rust-tools and #rust-lang. These four channels are the 'home' of the compiler team, libs team, tools team, and language team, respectively. Can you try indicating that in the descriptions, with links to the appropriate section on the team page?

Rust moderation team links to the team in the header as well as the description. Instead of the in-paragraph link to the team again perhaps 'contact' can be the link to their email.

Wow, you've written an entire contributing page too. Great start! Can you rename it to simply contributing.html and add a link to your modified navbar? A lot to think about here... I may be inclined to merge it as-is and circle back to think about it harder later. cc @nrc

"The comunity team is responsible for *helping to upholding* the code of conduct". Grammar typo.

Seems like internals.rust-lang.org should be included as well.

Well, this is pretty great. Thanks a lot :)

**Comment by brson** [comment]
"Our code of conduct must be observed in all matters relating to Rust online." This sentence is very important, but it's both strong ('must') and vague ('all matters'). I don't have a concrete suggestion right now but I'll think about it.

**Comment by efindlay** [comment]
Thanks @brson, Wouldn't contributing.md' too close to 'CONTRIBUTING.md'?

**Comment by brson** [comment]
Oh, huh. I didn't realize we had a 'contributing' file for the website. That is a drag. It sure seems like 'contributing.html' is a better name for the website page.

**Comment by efindlay** [comment]
I'm waiting on 'legendary dedication' link. Put the techwhirl link in temporarily.

**Comment by aturon** [comment]
@efindlay Thanks so much for doing this! I have a bunch of feedback, but think it might be easiest to give as a branch/PR against what you have here. Would that be ok with you?

**Comment by efindlay** [comment]
@aturon That's OK with me.

**Comment by brson** [comment]
Waiting on @aturon's feedback.

**Comment by aturon** [comment]
@efindlay Thanks again for the hard work on this! I've opened a PR with my editing pass: https://github.com/efindlay/rust-www/pull/1

cc @brson 

I focused only on the new community page, and haven't looked carefully at the contributing page yet.

**Comment by aturon** [comment]
BTW, once we've finished up the first round of feedback/edits, I think we should link to this PR in the Rust subreddit and/or discuss forum to get more eyeballs. I suspect most people aren't watching this repo closely.

File: community.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+[mod_team]: https://www.rust-lang.org/team.html#Moderation
+
+## Rust Development
+
+[How to contribute][contribute]
+
+According to [CodeTriage][codetriage], the Rust
+language has the most contributors of any programming language. If you
+want to join in the fun but aren't sure what to work on, please take
+a look at [how to contribute][contribute].
```
Comment (Manishearth) [inline community.md]: we link to [contribute] twice here, reads a bit strange

File: how_to_contribute.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
+---
+layout: default
+title: How To Contribute
```
Comment (Manishearth) [inline how_to_contribute.md]: So, I'm planning to have a generic "contribute to the Rust ecosystem" page as well. We probably should make this one more specific ("Contribute to the Rust compiler"); my page is not going to have major details about rustc itself.

Comment (Manishearth) [inline how_to_contribute.md]: (Or I can do this renaming later, really. No biggie)

**Comment by efindlay** [comment]
#### Mission 5, Documentation Page

1 Copied over documentation page.
2 Cleaned and fixed links.
3 Checked that there were no links to the old documentation page that needed to be changed.

(This should probably have been a new issue but I'm not sure how I could have done that.)

File: how_to_contribute.md (hunk 2 above)
Comment (brson) [inline how_to_contribute.md]: I'd prefer just one contributing page.

**Comment by brson** [comment]
I've posted the branch as it is now [to my own gh page](https://github.com/rust-lang/rust-www/issues/179) for easier review.

OK, so this covers [#2 - community page](https://github.com/rust-lang/rust-www/issues/176), [#3 - contributing](https://github.com/rust-lang/rust-www/issues/177), and now also [#5 - docs](https://github.com/rust-lang/rust-www/issues/179). This is a lot to digest :)

We agree there is going to be more follow up on the contributing page at least, so we don't need to make it (or anything really) perfect *right now*, as long is its presentable.

I'll try to get some more eyes on this, but some quick additional notes:

* Since this does add a contributing page and a docs page, I'd like to go ahead and stuff them in the navbar, even if we are going to redisign it still.
* 'We also have a twitter account' is terminated with a comma, should be a period.
* The reformatting paragraphs of IRC channels continue to contain inline dashes like a bulleted list. These are weird and I suggest removing them.
* The stylistic difference between h2 and h3 is *very* subtle. I'd like to try to make them more distinct.
* On the usergroups page, 'meetings are ... very informal'. Suggest removing 'very', filler word. Perhaps combine with the following sentence: 'meetings are usually informal and open to everyone', removing the 'monthly' assertion, since I'm not confident it generally holds.
* On the usergroups page, 'rust_www' i'd change to either 'rust-www' to match the repo name, or maybe just 'the website'.
* Try including the calendar again on the usergroups page, indicating that meetups are often listed there, perhaps even suggesting to contact the community team to add events.

The docs page needs to expose the docs for *all three channels*, since the front-page navbar links are going to disappear in favor of this page.

I think the 'community and getting help' section of the docs page can go away since we've got the community page also covering that content.

The faq link on the doc page goes to a weirdly formatted url of a page that doesn't exist.

Will you consider splitting the docs page into its own PR? It is going to require more work still than the community page, and I think it will be easier to review and move forward with them separate.

I still haven't looked closely at the contributing page, but it looks like a reasonable start.

**Comment by aturon** [comment]
I'd like to second @brson's request to have one PR per RWIB mission. I think that makes it a lot easier to get feedback and reach consensus, and allows us to start landing pieces independently.

**Comment by efindlay** [comment]
Thanks for all the feedback! Bear with me on the separate PR. I thought I made a separate PR but it didn't work out that way. I might need to do a little research to figure out how to sort it out.

**Comment by efindlay** [comment]
It looks like I need to make separate branches for each Mission ([Stack Overflow](http://stackoverflow.com/questions/8450036/how-to-open-multiple-pull-requests-on-github)). For Mission 3 Contributing Page and Mission 5 Documentation Page I will need to make PRs against new branches.

**Comment by edunham** [comment]
@efindlay, this is looking amazing! Sorting commits into separate PRs is one of the more complex "features" of GitHub. 

One fix would look something like this; easier if you want to mess around with fancy git ranges when cherry-picking:

```
$ git checkout 9f51aec7640ea7731779bf511e71ed9a9997259f #we're making a new branch that's even with rust-lang/rust-www master
$ git checkout -b mission3 # now we're on a branch named mission3
$ git cherry-pick abc123 # repeat for the hash of each commit you made for mission 3
$ git push origin mission3 # assuming your fork of the repo is the origin; `git config -l` to check
$ git checkout 9f51aec7640ea7731779bf511e71ed9a9997259f # back to rust-lang/rust-www master
$ git checkout -b mission5
$ git cherry-pick abc123 # again, repeat for each commit in your mission5 changes
$ git push origin mission5
$ git checkout abc123 # use the hash of the last commit in your mission 2 changes
$ git push --force origin master # overwrite the branch that this PR is from with only the mission 2 stuff
```
Rather than force-pushing to master, you could replace the final command with 
```
$ git checkout -b mission2
$ git push origin mission2
```
and then create a new mission2 PR from the branch you just pushed.

Now this PR would only show the mission 2 changes (since it's from your master branch, and you reset the master branch with the final command) and you'd have mission3 and mission5 branches to create separate PRs from. When you visit the main rust-lang/rust-www page after pushing mission3 and mission5, it'll say "you recently pushed branches" at the top and give you a button for creating those PRs.

**Comment by efindlay** [comment]
@edunham, thanks for you help. I'm still working on it ...

**Comment by brson** [comment]
Looks like this is superceded by https://github.com/rust-lang/rust-www/pull/201
